portal-service/src/com/liferay/portal/model/PortletInstance.java

```diff
@@ -114,6 +133,48 @@ public String toString() {
 		return getPortletInstanceKey();
 	}
 
+	private static FullInstanceId _buildFullInstanceId(String fullInstanceId) {
+		if (fullInstanceId == null) {
+			throw new InvalidParameterException(
+				"The fullInstanceId must not be null.");
+		}
+
+		String[] tokens = StringUtil.split(
+			fullInstanceId, StringPool.UNDERLINE);
+
+		int count = StringUtil.count(fullInstanceId, StringPool.UNDERLINE);
```

Comment:
In that case the split it's not enough, as it takes two tokens, tokenizes it right , but the fullInstanceId is still wrong